Adding Happiness Metric

mmilkowski36 opened the pull request.

nkremerh reviewed (changes requested):
Good first pass. Let's button up a few remaining pieces:

1. I am not sure why importing the sugarscape module is necessary in the agent.
2. Same with passing in a meanWealthList to the agent constructor.
3. Avoid shortening variable and function names (e.g. meanWealthL, calcFamilyHappiness). Make those names verbose.
4. Remove the two ethical decisionmaking models out of agent.py since those have been moved to ethics.py.
5. Line 47 of sugarscape.py removes the totalAccumulation data field which we want to keep.

mmilkowski36 commented:
Sugarscape and meanWealthList is to pass average wealth into agent.py
I have removed the ethics sections from agent.py, and changed sugarscape.py so totalAccumilation is on the same line as last time

WillemHueffed commented:
Setting headless mode to true and debug mode to false breaks simulation:
"AttributeError: 'NoneType' object has no attribute 'environment'"
Config file:
{
    "__README__": "Default values for Sugarscape simulation provided here. Details can be found in the README.",
    "agentAggressionFactor": [1, 1],
    "agentBaseInterestRate": [0.05, 0.10],
    "agentEthicalFactor": [1, 1],
    "agentEthicalTheory": "none",
    "agentFemaleInfertilityAge": [40, 50],
    "agentFemaleFertilityAge": [12, 15],
    "agentFertilityFactor": [1, 1],
    "agentImmuneSystemLength": 35,
    "agentInheritancePolicy": "children",
    "agentLendingFactor": [1, 1],
    "agentLoanDuration": [5, 5],
    "agentLookaheadFactor": [1, 10],
    "agentMaleInfertilityAge": [50, 60],
    "agentMaleFertilityAge": [12, 15],
    "agentMaleToFemaleRatio": 1,
    "agentMaxAge": [60, 100],
    "agentMaxFriends": [5, 10],
    "agentMovement": [1, 6],
    "agentReplacements": 0,
    "agentSpiceMetabolism": [1, 4],
    "agentStartingSpice": [10, 40],
    "agentStartingSugar": [10, 40],
    "agentStartingQuadrants": [1, 2, 3, 4],
    "agentSugarMetabolism": [1, 4],
    "agentTagStringLength": 11,
    "agentTradeFactor": [1, 1],
    "agentVision": [1, 6],
    "debugMode": false,
    "diseaseAggressionPenalty": [-1, 1],
    "diseaseFertilityPenalty": [-1, 1],
    "diseaseMovementPenalty": [0, 0],
    "diseaseSpiceMetabolismPenalty": [1, 3],
    "diseaseSugarMetabolismPenalty": [1, 3],
    "diseaseTagStringLength": [11, 21],
    "diseaseVisionPenalty": [-1, 1],
    "environmentHeight": 50,
    "environmentMaxCombatLoot": 1,
    "environmentMaxSpice": 4,
    "environmentMaxSugar": 4,
    "environmentMaxTribes": 3,
    "environmentPollutionDiffusionDelay": 0,
    "environmentSeasonalGrowbackDelay": 0,
    "environmentSeasonInterval": 0,
    "environmentSpiceConsumptionPollutionFactor": 0,
    "environmentSpiceProductionPollutionFactor": 0,
    "environmentSpiceRegrowRate": 1,
    "environmentSugarConsumptionPollutionFactor": 0,
    "environmentSugarProductionPollutionFactor": 0,
    "environmentSugarRegrowRate": 1,
    "environmentWidth": 50,
    "headlessMode": true,
    "logfile": "log.json",
    "profileMode": false,
    "seed": -1,
    "startingAgents": 500,
    "startingDiseases": 50,
    "timesteps": 200
}

nkremerh commented:
This looks like the history is back where it should be. Let's get this re-tested. I'll make another review pass. After any changes (if there are any), squash the commits. It will be good to merge at that point.

WillemHueffed commented:
This looks good, there are some minor whitespace differences that could get fixed. Config bug is fixed.  History needs squashing, then its ready for commit. Well done Maria! :)

nkremerh commented:
Let's plan to have this fully ready to merge by next Tuesday's meeting. Follow up with me offline to give me a status update on where this PR is at and what is still left to be done to get it over the finish line.

nkremerh commented:
This looks great! Let's get one last peer code review in, then it's good to merge.

lucas-vorkpork commented:
Looks great to me, this will take the research in an interesting direction!

nkremerh commented:
Alright, @mmilkowski36 squash the commits for merge.

mmilkowski36 commented:
squashed history :)
